OpenStack-Rabbit-Consumer/rabbit_consumer/message_consumer.py

```diff
@@ -254,6 +254,31 @@ def on_message(message: rabbitpy.Message) -> None:
     message.ack()
 
 
+def generate_login_str(config: ConsumerConfig) -> str:
+    """
+    Generates the login string for the rabbit connection.
+    """
+    if not config.rabbit_hosts:
+        raise ValueError("No rabbit hosts provided")
+
+    if not isinstance(config.rabbit_hosts, str):
+        raise ValueError("Rabbit hosts must be a comma separated string of hosts")
+
+    connect_str = "amqp://"
+
+    for host in config.rabbit_hosts.split(","):
+        host = host.strip()
+        connect_str += f"{config.rabbit_username}:{config.rabbit_password}@{host}:{config.rabbit_port},"
+
+    # Trim the trailing comma
+    connect_str = connect_str[:-1]
+
+    sanitised_connect_str = connect_str.replace(config.rabbit_password, "<password>")
+    logger.debug("Connecting to rabbit with: %s", sanitised_connect_str)
```

Comment:
hmm - this just feels wrong to me. You're unnecessarily risking the password being leaked  

I'd prefer you construct a variable separate that we can then output
```suggestion
    connect_str, debug_str = "amqp://"
    
    for host in config.rabbit_hosts.split(","):
        host = host.strip()
        connect_str += f"{config.rabbit_username}:{config.rabbit_password}@{host}:{config.rabbit_port},"
        debug_str += f"<username>:<password>@{host}:{config.rabbit_port},"

    # Trim the trailing comma
    connect_str = connect_str[:-1]
    debug_str = debug_str[:-1]

    logger.debug("Connecting to rabbit with: %s", debug_str)
```